### PR TITLE
[wip] Break up Test Workflows

### PR DESCRIPTION
**Pull request by rawkintrevo**

### CODE REVIEW

**Comment by rawkintrevo** [comment]
ALL. this is _not_ ready to merge yet. I need to update the paths, when it is ready to merge, this will make it so it only runs the test if certain files were updates (e.g. don't test CLDK notebooks on a doc merge)

@deanwampler one of the cldk notebooks got pushed to 8b, i rolled it back to 3b. not sure if that was on purpose or by accident, if you need to put it back to 8b, please open a new ticket and try it. that CLDK pipeline is still pushing the limit. 

@adampingel / @fayvor lmk your feels on names of things

File: .github/workflows/cldk_workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  push:
+    branches:
+      - main
+#    paths:
+#      - '.github/notebook_lists/cldk_notebooks.txt'
+  pull_request:
+    branches:
+      - main
+#    paths:
+#      - '.github/notebook_lists/cldk_notebooks.txt'
```
Comment (rawkintrevo) [inline .github/workflows/cldk_workflow.yaml:13]: these lines will specify to only run the test if one of the files listed here have been changed, i will update with the paths of the notebooks in the CLDK directory

File: recipes/CodeLLM_DevKit/generate_unit_tests.ipynb [hunk 2]
```diff
@@ -166,7 +166,7 @@
    "source": [
     "#### Step 3: Define a function to call the LLM\n",
     "\n",
-    "As before in the [Code Sumarization](./code_summarization.ipynb) recipe, we use Ollama with Granite Code 8b Instruct."
+    "As before in the [Code Sumarization](./code_summarization.ipynb) recipe, we use Ollama with Granite Code 3b Instruct."
```
Comment (rawkintrevo) [inline recipes/CodeLLM_DevKit/generate_unit_tests.ipynb:169]: @deanwampler here and right below

File: recipes/Getting_Started_with_Granite_Code.ipynb [hunk 3]
```diff
@@ -32,444 +32,12 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": null,
```
Comment (rawkintrevo) [inline recipes/Getting_Started_with_Granite_Code.ipynb:35]: this notebook got committed without having the outputs cleared

File: .github/workflows/vanilla_workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    paths:
+      - '.github/notebook_lists/vanilla_notebooks.txt'
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '.github/notebook_lists/vanilla_notebooks.txt'
+
+jobs:
+  test-torch-notebooks:
```
Comment (rawkintrevo) [inline .github/workflows/vanilla_workflow.yaml:16]: updated `torch` to vanilla already

File: .github/notebook_lists/torch_notebooks.txt
Comment (rawkintrevo) [inline .github/notebook_lists/torch_notebooks.txt:1]: deleted this file

File: .github/workflows/ollama_workflow.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+name: CI for Ollama Notebooks
```
Comment (fayvor) [inline .github/workflows/ollama_workflow.yaml:1]: Note that the objective now is to run all (code) notebooks from Replicate by default. So Ollama testing should be confined to the "Kitchen", and maybe some repos where we don't have the necessary models on Replicate.

Comment (rawkintrevo) [reply .github/workflows/ollama_workflow.yaml:1]: cool, once those get moved we can drop or move this workflow

File: .github/workflows/vanilla_workflow.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
+name: CI for Vanilla Notebooks
```
Comment (fayvor) [inline .github/workflows/vanilla_workflow.yaml:1]: Good name for the cooking theme :)

**Comment by fayvor** [comment]
Looks good to me so far. Thanks for taking care of this!

File: .github/workflows/ollama_workflow.yaml (hunk 5 above)
Comment (adampingel) [reply .github/workflows/ollama_workflow.yaml:1]: Agreed with the point.  I'll write a ticket for this objective.

File: .github/workflows/vanilla_workflow.yaml (hunk 6 above)
Comment (adampingel) [reply .github/workflows/vanilla_workflow.yaml:1]: bravo

**Review by adampingel — APPROVED** [review]
We should follow up with some documentation, including here:

https://github.com/ibm-granite-community/community/wiki/Automated-Validation

**Comment by rawkintrevo** [comment]
Why did this get merged??

It was marked [WIP] for a reason, let's discuss git hygiene on our call, including merging PRs after hours

**Pull request merged**